### PR TITLE
[Quantum RPG] Quantopedia

### PR DESCRIPTION
**Pull request by dstrain115**
- Adds an in-game help option for a quantopedia.
- When you find the book or item that has info on the type of enemy you are facing, you can type 'q' and learn about it.

### CODE REVIEW

File: unitary/examples/quantum_rpg/game_state.py [hunk 1]
```diff
@@ -42,6 +43,18 @@ def __init__(
...
+        if _QUANTOPEDIA_KEY not in self.state_dict:
+            self.state_dict[_QUANTOPEDIA_KEY] = "0"
+        quantopedia_num = int(self.state_dict[_QUANTOPEDIA_KEY]) | num
+        self.state_dict[_QUANTOPEDIA_KEY] = str(quantopedia_num)
+
+    def has_quantopedia(self, num):
+        if _QUANTOPEDIA_KEY not in self.state_dict:
+            return False
+        quantopedia_num = int(self.state_dict[_QUANTOPEDIA_KEY])
+        return (quantopedia_num // num) % 2 == 1
```
Comment (losos0) [inline unitary/examples/quantum_rpg/game_state.py:56]: `quantopedia_num & num != 0`

File: unitary/examples/quantum_rpg/npcs.py [hunk 2]
```diff
@@ -67,6 +67,14 @@ def npc_action(self, battle, **kwargs) -> str:
         action_choice = random.random()
         return self.act_on_enemy_qubit(enemy_qubit, action_choice, **kwargs)
 
+    def quantopedia_index(self) -> int:
+        """Bit to use for quantopedia.  Should be power of two."""
+        return 1
```
Comment (losos0) [inline unitary/examples/quantum_rpg/npcs.py:72]: Maybe 0 would be a better default? (impossible to learn about it)

**Review by losos0 — COMMENTED** [review]
I guess it's lacking a call to set_quantopedia to be usable, but what's written so far seems OK.

File: unitary/examples/quantum_rpg/game_state.py (hunk 1 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/game_state.py]: Done.

File: unitary/examples/quantum_rpg/npcs.py (hunk 2 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/npcs.py]: Done.  Also, good point, I have added docstrings all around and have also added constants for each bit.

**Comment by dstrain115** [comment]
FYI, set_quantopedia is called when reading books in oxtail, 
[]see here](https://github.com/quantumlib/unitary/pull/127/files#diff-fe847231b3c299f4d33e2b2b16ffe286acbdc2ed62e01ffb1a76663478cb7c88R77)

**Pull request merged**